
From: Jesse Barnes <jbarnes@engr.sgi.com>

Fix up ia64 specific memory map init function in light of Dave's
memmap_init cleanups.

Signed-off-by: Jesse Barnes <jbarnes@sgi.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/ia64/mm/init.c        |    6 ++++--
 25-akpm/include/asm-ia64/pgtable.h |    2 +-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff -puN arch/ia64/mm/init.c~dont-pass-mem_map-into-init-functions-ia64-fix arch/ia64/mm/init.c
--- 25/arch/ia64/mm/init.c~dont-pass-mem_map-into-init-functions-ia64-fix	2004-08-09 22:02:11.905819088 -0700
+++ 25-akpm/arch/ia64/mm/init.c	2004-08-09 22:02:11.909818480 -0700
@@ -435,14 +435,16 @@ virtual_memmap_init (u64 start, u64 end,
 }
 
 void
-memmap_init (struct page *start, unsigned long size, int nid,
-	     unsigned long zone, unsigned long start_pfn)
+memmap_init (unsigned long size, int nid, unsigned long zone,
+	     unsigned long start_pfn)
 {
 	if (!vmem_map)
 		memmap_init_zone(size, nid, zone, start_pfn);
 	else {
+		struct page *start;
 		struct memmap_init_callback_data args;
 
+		start = pfn_to_page(start_pfn);
 		args.start = start;
 		args.end = start + size;
 		args.nid = nid;
diff -puN include/asm-ia64/pgtable.h~dont-pass-mem_map-into-init-functions-ia64-fix include/asm-ia64/pgtable.h
--- 25/include/asm-ia64/pgtable.h~dont-pass-mem_map-into-init-functions-ia64-fix	2004-08-09 22:02:11.906818936 -0700
+++ 25-akpm/include/asm-ia64/pgtable.h	2004-08-09 22:02:11.910818328 -0700
@@ -520,7 +520,7 @@ do {											\
 #  ifdef CONFIG_VIRTUAL_MEM_MAP
   /* arch mem_map init routine is needed due to holes in a virtual mem_map */
 #   define __HAVE_ARCH_MEMMAP_INIT
-    extern void memmap_init (struct page *start, unsigned long size, int nid, unsigned long zone,
+    extern void memmap_init (unsigned long size, int nid, unsigned long zone,
 			     unsigned long start_pfn);
 #  endif /* CONFIG_VIRTUAL_MEM_MAP */
 # endif /* !__ASSEMBLY__ */
_
